File: kyungyeul/algorithm1/200/스택/stack.py
import sys


def push(x):
    stack.append(x)


def pop():
    if len(stack) == 0:
        return -1
    else:
        return stack.pop()


def size():
    return len(stack)


def empty():
    if len(stack) == 0:
        return 1
    else:
        return 0


def top():
    if len(stack) == 0:
        return -1
    else:
        return stack[-1]


n = int(sys.stdin.readline())
stack = []

for _ in range(n):
    commend = sys.stdin.readline().split()
    if commend[0] == "push":
        push(commend[1])
    elif commend[0] == "pop":
        print(pop())
    elif commend[0] == "size":
        print(size())
    elif commend[0] == "empty":
        print(empty())
    elif commend[0] == "top":
        print(top())

File: kyungyeul/algorithm1/200/스택/stack.js
const stack = [];
let top = 0;
let answer = "";

const result = {
  push: (x) => {
    stack.push(x);
  },
  pop: () => {
    if (stack.length == 0) {
      return -1;
    } else {
      return stack.pop();
    }
  },
  size: () => {
    return stack.length;
  },
  empty: () => {
    if (stack.length == 0) {
      return 1;
    } else {
      return 0;
    }
  },
  top: () => {
    if (stack.length == 0) {
      return -1;
    } else {
      return stack[stack.length - 1];
    }
  },
};

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  const n = Number(input[0]);
  for (let i = 1; i <= n; i++) {
    let commend = input[i].split(" ");
    if (commend[0] == "push") {
      result.push(commend[1]);
    } else if (commend[0] == "pop") {
      console.log(result.pop());
    } else if (commend[0] == "size") {
      console.log(result.size());
    } else if (commend[0] == "empty") {
      console.log(result.empty());
    } else if (commend[0] == "top") {
      console.log(result.top());
    }
  }
  process.exit();
});
